src/main/scala/rna16s.scala

```diff
@@ -36,27 +36,34 @@ case object bio4jTaxonomyBundle extends AnyBio4jDist {
     TitanEdge, EdgeLabelMaker
   ]
 
-  def checkAncestors(id: String, ancestors: Set[String]): Boolean = {
-    // Java to Scala
-    def optional[T](jopt: java.util.Optional[T]): Option[T] =
-      if (jopt.isPresent) Some(jopt.get) else None
-
-    @scala.annotation.tailrec
-    def checkAncestors_rec(node: TaxonNode): Boolean =
-      optional(node.ncbiTaxonParent_inV) match {
-        case None => false
-        case Some(parent) =>
-          if (ancestors.contains( parent.id() )) true
-          else checkAncestors_rec(parent)
-      }
-
-    optional(graph.nCBITaxonIdIndex.getVertex(id))
-      .map(checkAncestors_rec)
-      .getOrElse(false)
+  implicit class IdOps(val id: String) extends AnyVal {
```

Comment:
I don't see much point in this dancing over adding one method to _String_... But whatevs..